scripts/regression_test/extension_regression_test.yml

```diff
@@ -0,0 +1,191 @@
+    displayName: "Rerun tests"
+  - bash: |
+      publishErrorModules='false'
+      if [[ -f '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_error_modules_$(Instance_idx).txt' ]]; then
+        publishErrorModules='true'
+      fi
+      echo "##vso[task.setvariable variable=publishErrorModules]$publishErrorModules"
+
+      publishFailureTests='false'
+      if [[ -f '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_failure_tests_$(Instance_idx).txt' ]]; then
+        publishFailureTests='true'
+      fi
+      echo "##vso[task.setvariable variable=publishFailureTests]$publishFailureTests"
+    condition: succeededOrFailed()
+    displayName: "Prepare publishing results"
+  - task: PublishTestResults@2
+    condition: succeededOrFailed()
+    inputs:
+      testResultsFiles: '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_*.xml'
+      testRunTitle: 'CLI Regression test results of instance $(Instance_idx)'
+  - task: PublishBuildArtifacts@1
+    condition: and(succeededOrFailed(), eq(variables.publishErrorModules, 'true'))
+    inputs:
+      PathtoPublish: '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_error_modules_$(Instance_idx).txt'
+      ArtifactName: 'error_modules'
+      publishLocation: 'Container'
+  - task: PublishBuildArtifacts@1
+    condition: and(succeededOrFailed(), eq(variables.publishFailureTests, 'true'))
+    inputs:
+      PathtoPublish: '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_failure_tests_$(Instance_idx).txt'
+      ArtifactName: 'failure_tests'
+      publishLocation: 'Container'
+
+- job: ResultSummary
+  displayName: CLI Extension Regression Test Result Summary
+  dependsOn: CLIExtensionRegressionTest
+  condition: succeededOrFailed()
+  pool:
+    name: ${{ variables.ubuntu_pool }}
+  steps:
+    - task: AzureCLI@1
+      displayName: 'Result Summary'
+      inputs:
+        azureSubscription: 'Azure CLI release'
+        scriptLocation: inlineScript
+        inlineScript: |
+          set -ev
+
+          # git config
+          git clone https://github.com/Azure/azure-cli-extensions.git
+          cd azure-cli-extensions
+
+          # git config
+          GITHUB_TOKEN=$(az keyvault secret show --vault-name kv-azuresdk --name azclibot-pat --query value -otsv)
+          git config --global user.email "AzPyCLI@microsoft.com"
+          git config --global user.name "Azure CLI Team"
+          git remote add azclibot https://azclibot:${GITHUB_TOKEN}@github.com/azclibot/azure-cli-extensions.git
+
+          git fetch azclibot
+          REGRESSION_BRANCHES=$(curl -H "Authorization: token ${GITHUB_TOKEN}" https://api.github.com/repos/azclibot/azure-cli-extensions/git/matching-refs/heads/regression_test_$(Build.BuildId)_ | jq '.[] | .ref')
+
+          for branch in ${REGRESSION_BRANCHES[@]}
+          do
+            branch=${branch##*/}
+            branch=${branch%\"*}
```

Comment:
Can we add an example to better understand these code?

Reply:
Done